### PR TITLE
Added params and sorter descriptions for all sorters

### PR DESCRIPTION
**Pull request by alejoe91**
- HS2:
  - changed `detection_threshold` --> `detect_threshold`
  - @mhhennig @colehurwitz can you fill in the last parameter descriptions?

- TDC:
  - changed some parameters so that they are the same among sorters: `freq_min` / `freq_max` / 'detect_threshold` / `detect_sign`
  - @samuelgarcia check params description

- Waveclus: @ferchaure can you fill in the last parameters and check that the ones I input are correct?

### CODE REVIEW

**Comment by ferchaure** [comment]
Done in my branch [waveclus.py](https://github.com/ferchaure/spikesorters/blob/master/spikesorters/waveclus/waveclus.py). It's easier than open a new pull request

File: spikesorters/herdingspikes/herdingspikes.py [hunk 1]
```diff
@@ -119,50 +207,3 @@ def _run(self, recording, output_folder):
     @staticmethod
     def get_result_from_folder(output_folder):
         return se.HS2SortingExtractor(file_path=Path(output_folder) / 'HS2_sorted.hdf5', load_unit_info=True)
-
-
-HerdingspikesSorter._default_params = {
-    # core params
```
Comment (samuelgarcia) [inline spikesorters/herdingspikes/herdingspikes.py:125]: I think I preferd this way thanb then nested as you did.

File: spikesorters/basesorter.py [hunk 2]
```diff
@@ -97,8 +99,12 @@ def __init__(self, recording=None, output_folder=None, verbose=False,
         self.delete_folders = delete_output_folder
 
     @classmethod
-    def default_params(self):
-        return copy.deepcopy(self._default_params)
+    def default_params(cls):
+        return copy.deepcopy(cls._default_params)
+
```
Comment (samuelgarcia) [inline spikesorters/basesorter.py:104]: cool

File: spikesorters/hdsort/hdsort.py [hunk 3]
```diff
@@ -47,6 +47,26 @@ class HDSortSorter(BaseSorter):
...
+        'add_if_nearer_than': "Minimum distance to add electrode to an electrode group",
+        'max_distance_within_group': "Maximum distance within an electrode group",
+        'n_pc_dims': "Number of principal components dimensions to perform initial clustering",
+        'chunk_size': "Chunk size in number of frames for template-matching",
+        'loop_mode': "Loop mode: 'loop', 'local_parfor', 'grid' (requires a grid architecture)",
+    }
+
+    sorter_description = """HDSort is a template-matching spike sorter designed for high density micro-electrode arrays. 
+    For more information see https://doi.org/10.1152/jn.00803.2017"""
+
```
Comment (samuelgarcia) [inline spikesorters/hdsort/hdsort.py:69]: This is a cool idea.
I think we should use also this generate `_doc_` for functions lanucher.

File: spikesorters/herdingspikes/herdingspikes.py (hunk 1 above)
Comment (alejoe91) [reply spikesorters/herdingspikes/herdingspikes.py:125]: why? it is like this for all other sorters and it's the same in the end , no?

File: spikesorters/sorterlist.py [hunk 4]
```diff
@@ -136,6 +137,56 @@ def get_default_params(sorter_name_or_class):
...
+    '''
+    if isinstance(sorter_name_or_class, str):
+        SorterClass = sorter_dict[sorter_name_or_class]
+    elif sorter_name_or_class in sorter_full_list:
+        SorterClass = sorter_name_or_class
+    else:
+        raise (ValueError('Unknown sorter'))
+
+    return SorterClass.params_description()
+
```
Comment (samuelgarcia) [inline spikesorters/sorterlist.py:163]: Cool.

File: spikesorters/sorterlist.py [hunk 5]
```diff
@@ -136,6 +137,56 @@ def get_default_params(sorter_name_or_class):
...
+
+def get_sorter_description(sorter_name_or_class):
```
Comment (samuelgarcia) [inline spikesorters/sorterlist.py:165]: Cool.

File: spikesorters/tridesclous/tridesclous.py [hunk 6]
```diff
@@ -103,6 +120,16 @@ def _run(self, recording, output_folder):
...
+        # parse params to tdc names
+        params['highpass_freq'] = params['freq_min']
+        params['lowpass_freq'] = params['freq_max']
+        params['relative_threshold'] = params['detect_threshold']
+        if params['detect_sign'] == -1:
+            params['peak_sign'] = '-'
+        elif params['detect_sign'] == 1:
+            params['peak_sign'] = '+'
+        del params['freq_min'], params['freq_max'], params['detect_sign'], params['detect_threshold']
+
```
Comment (samuelgarcia) [inline spikesorters/tridesclous/tridesclous.py:132]: please do this in make_nested_tdc_params

File: spikesorters/herdingspikes/herdingspikes.py (hunk 1 above)
Comment (samuelgarcia) [reply spikesorters/herdingspikes/herdingspikes.py:125]: yes but when the params list is very long.
it breaks the readablity of the code.
as you want it do not matter a lot.

**Comment by samuelgarcia** [comment]
This is Ok for me.

I think we could also hack this parameters description to enhance the `__doc__` for `run_XXXX`.

File: spikesorters/tridesclous/tridesclous.py [hunk 7]
```diff
@@ -159,22 +177,26 @@ def get_result_from_folder(output_folder):
 
 
 def make_nested_tdc_params(tdc_dataio, chan_grp,
```
Comment (samuelgarcia) [inline spikesorters/tridesclous/tridesclous.py:179]: perfect

**Pull request merged**